session_test.go

```diff
@@ -117,3 +117,62 @@ func TestAddStreamAfterSessionClose(t *testing.T) {
 	ustr.EXPECT().CancelRead(sessionCloseErrorCode)
 	sess.addIncomingUniStream(ustr)
 }
+
+func TestOpenStreamAfterSessionClose(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockSess := NewMockStreamCreator(ctrl)
+	mockSess.EXPECT().Context().Return(context.WithValue(context.Background(), quic.ConnectionTracingKey, uint64(1337)))
+	wait := make(chan struct{})
+	mockSess.EXPECT().OpenStreamSync(gomock.Any()).DoAndReturn(func(context.Context) (quic.Stream, error) {
+		str := NewMockStream(ctrl)
+		str.EXPECT().CancelRead(sessionCloseErrorCode)
+		str.EXPECT().CancelWrite(sessionCloseErrorCode)
+		<-wait
+		return str, nil
+	})
+
+	sess := newSession(42, mockSess, newMockRequestStream(ctrl))
+
+	errChan := make(chan error, 1)
+	go func() {
+		_, err := sess.OpenStreamSync(context.Background())
+		errChan <- err
+	}()
+
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Use `require.Eventually`?

Ditto in other places that use `time.Sleep`.

Reply:
Refactored such that we don't need to sleep at all :)